### PR TITLE
Use native ExpandTypeVisitor for expanding type aliases as well

### PR DESCRIPTION
**Pull request by ilevkivskyi**
This used to be a source of code duplication, and also using the "native" visitor will allow us to implement new features for type aliases easily (e.g. variadic type aliases, that I am going to do in next PR).

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Looking at mypy-primer:

* The pytorch/vision error is [here](https://github.com/pytorch/vision/blob/0370134359268666bb1de9b53c1e9e64f1d4cde7/torchvision/models/_api.py#L125). The new error is correct; `x[1]` is a module and `ModuleType.__file__` is str | None. Mypy got better at type inference here.
* The pandas hits are [here](https://github.com/apache/spark/blob/d157b2d9a71c8370612342b298f0e6757c9b9b78/python/pyspark/pandas/supported_api_gen.py#L160). Very similar situation where mypy now infers a more precise type for an `inspect.getmembers` call.
* steam.py is [here](https://github.com/Gobot1234/steam.py/blob/1f31dd3d5c77ac1babc9451af2638c2c9aa8890c/steam/state.py#L2696), another `inspect.getmembers` call. The code is buggy and the mypy error seems like a false positive, but again it looks like this PR infers more precise types. The [second steam.py hit](https://github.com/Gobot1234/steam.py/blob/1f31dd3d5c77ac1babc9451af2638c2c9aa8890c/steam/_gc/state.py#L82) looks similar.

In conclusion, the PR does a better job of inferring types around `inspect.getmembers`.

**Comment by AlexWaygood** [comment]
For context, the more precise types we currently have for `inspect.getmembers()` in typeshed were added quite recently, in https://github.com/python/typeshed/pull/9891 (by @Gobot1234, the author of `steam.py`). I noted in that PR thread that mypy didn't seem able to properly understand the new types we were adding. It's great that it now can!

Could we add a test case for functions where a parameter is annotated with an alias to a generic TypeGuard?

**Comment by ilevkivskyi** [comment]
Yeah, it looks like a bug (or maybe not) in `TypeTranslator`, it doesn't translate the `.type_guard` in `visit_callable_type()`. Since `ExpandTypeVisitor` overrides it with some non-trivial logic that also includes `.type_guard.accept(self)` it worked there, but `InstantiateAliasVisitor` simply inherited the method before.

For now I will just add a test.

**Review by hauntsaninja — APPROVED** [review]
Thank you! And thanks Jelle for looking through the primer diff
